### PR TITLE
🧪 [testing improvement] add error handling coverage for paper edit page

### PR DESCRIPTION
**Pull request by is0692vs**
🎯 **What:** Adds unit tests for the error handling logic in the `PaperEditPage` component when fetching paper details fails with unexpected errors (both `Error` instances and non-Error primitives).

📊 **Coverage:**
- When `apiFetch` throws an `Error` instance, tests that `err.message` is correctly extracted and displayed in the UI.
- When `apiFetch` throws a non-Error (e.g. string), tests that the fallback Japanese message `"予期せぬエラーが発生しました"` is successfully displayed.

✨ **Result:** Enhanced test coverage for edge case failure scenarios in `PaperEditPage`, ensuring the UI properly communicates unexpected network and fetch failures.

---
*PR created automatically by Jules for task [17082327013811816704](https://jules.google.com/task/17082327013811816704) started by @is0692vs*

<details><summary><h3>Greptile Summary</h3></summary>

本 PR は `PaperEditPage` コンポーネントのデータ取得失敗時におけるエラーハンドリングのユニットテストを追加します。`apiFetch` が `Error` インスタンスをスローした場合と、文字列などの非 Error プリミティブをスローした場合の 2 つのエッジケースをカバーしており、既存のエラーハンドリング実装（`page.tsx` 132行目）が UI に正しく反映されることを検証します。

- `Error` インスタンス投入時：`err.message` が画面に表示されることを確認
- 非 Error プリミティブ投入時：フォールバックメッセージ `"予期せぬエラーが発生しました"` が表示されることを確認
- `beforeEach` が未使用のままインポートされている（スタイル上の軽微な問題）
- `package-lock.json` に `fsevents` の `"dev": true` 削除という無関係な変更が混入しており、本 PR のスコープ外の変更として要確認
</details>

<h3>Confidence Score: 4/5</h3>

- テストロジック自体は正しく動作しますが、`package-lock.json` への無関係な変更が混入しているため、マージ前に確認が必要です。
- 追加されたテストは実装のエラーハンドリングロジックを正確に検証しており、機能的に問題はありません。ただし、`beforeEach` の未使用インポートと `package-lock.json` への意図しないと思われる変更（`fsevents` の `"dev": true` 削除）が含まれています。後者は本番依存関係に影響する可能性があるため確認が推奨されます。
- `package-lock.json` の変更が意図的かどうかを確認してください。

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| apps/web/src/app/papers/[id]/edit/__tests__/page.test.tsx | PaperEditPage のエラーハンドリングに関する新規テストファイル。`Error` インスタンスと非 Error 値（文字列等）を `apiFetch` がスローする 2 つのシナリオをカバー。`beforeEach` の未使用インポートが 1 件あり。 |
| package-lock.json | `fsevents@2.3.2` の `"dev": true` フィールドが削除されており、本 PR のテスト追加とは無関係な変更が混入している。意図しない変更の可能性が高い。 |

</details>

</details>

<h3>Sequence Diagram</h3>

```mermaid
sequenceDiagram
    participant Test as テスト
    participant Component as PaperEditPage
    participant apiFetch as apiFetch (Mock)
    participant UI as UI レンダリング

    Test->>apiFetch: mockRejectedValueOnce(new Error("Network fetch failed"))
    Test->>Component: render()
    Component->>UI: ローディングスピナー表示 (loading=true)
    Component->>apiFetch: apiFetch("/api/papers/paper-1")
    apiFetch-->>Component: throw Error("Network fetch failed")
    Component->>Component: setError("Network fetch failed")
    Component->>Component: setLoading(false)
    Component->>UI: エラーメッセージ "Network fetch failed" 表示
    Test->>UI: waitFor → getByText("Network fetch failed") ✅

    Test->>apiFetch: mockRejectedValueOnce("Unexpected string error")
    Test->>Component: render()
    Component->>UI: ローディングスピナー表示 (loading=true)
    Component->>apiFetch: apiFetch("/api/papers/paper-1")
    apiFetch-->>Component: throw "Unexpected string error" (非Errorプリミティブ)
    Component->>Component: setError("予期せぬエラーが発生しました")
    Component->>Component: setLoading(false)
    Component->>UI: フォールバックエラーメッセージ表示
    Test->>UI: waitFor → getByText("予期せぬエラーが発生しました") ✅
```

<details><summary>Prompt To Fix All With AI</summary>

`````markdown
This is a comment left during a code review.
Path: apps/web/src/app/papers/[id]/edit/__tests__/page.test.tsx
Line: 4

Comment:
**未使用の `beforeEach` インポート**

`beforeEach` がインポートされていますが、テストファイル内のどこにも使用されていません。不要なインポートはコードの可読性を下げるため、削除することを推奨します。

```suggestion
import { afterEach, describe, expect, it, vi } from "vitest";
```

How can I resolve this? If you propose a fix, please make it concise.

---

This is a comment left during a code review.
Path: package-lock.json
Line: 10592-10596

Comment:
**無関係な `fsevents` の変更**

テストファイルの追加とは無関係に、`fsevents@2.3.2` から `"dev": true` フィールドが削除されています。

この変更により `fsevents`（macOS 専用のファイルシステムイベントモジュール）が開発依存から通常依存に昇格し、本番ビルドやインストール時に不要なモジュールが含まれてしまう可能性があります。

意図的な変更でないのであれば、このファイルを元のバージョンに戻すことを推奨します。ローカルで `npm install` を実行した環境（例：macOS）と CI 環境の差異によって生じた可能性があります。

How can I resolve this? If you propose a fix, please make it concise.
`````

</details>

<sub>Last reviewed commit: ["test: add error hand..."](https://github.com/hiroki-org/openshelf/commit/06bfac3ee6d8295e2c97a04d9c4f2bbd79e1fa26)</sub>

> Greptile also left **2 inline comments** on this PR.

### CODE REVIEW

File: apps/web/src/app/papers/[id]/edit/__tests__/page.test.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import { cleanup, render, screen, waitFor } from "@testing-library/react";
+import { afterEach, beforeEach, describe, expect, it, vi } from "vitest";
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply apps/web/src/app/papers/[id]/edit/__tests__/page.test.tsx]: 対応しました: `beforeEach` の未使用インポートを削除しました。影響範囲: `apps/web/src/app/papers/[id]/edit/__tests__/page.test.tsx` の import 行のみです。検証: `npm run -w apps/web test -- "src/app/papers/[id]/edit/__tests__/page.test.tsx"` で 2 tests passed を確認済みです。

File: apps/web/src/app/papers/[id]/edit/__tests__/page.test.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+import PaperEditPage from "../page";
+import { apiFetch } from "@/lib/api";
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply apps/web/src/app/papers/[id]/edit/__tests__/page.test.tsx]: 対応しました: 同一指摘のため `beforeEach` の未使用インポートを削除済みです。影響範囲: テストファイルの import 文のみです。検証: `npm run -w apps/web test -- "src/app/papers/[id]/edit/__tests__/page.test.tsx"` でテスト通過を確認しました。
